### PR TITLE
[MISC] Fix missing IP errors

### PR DESCRIPTION
**Pull request by dragomirp**
Should fix issues happening in the data-integrator tests: https://github.com/canonical/data-integrator/actions/runs/10625135518/job/29454710269#step:28:350

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -985,11 +985,6 @@ def update_client_connection_info(self):
         for relation in self.model.relations.get(CLIENT_RELATION_NAME, []):
             self.client_relation.update_connection_info(relation)
 
-    @property
-    def leader_ip(self) -> str:
-        """Gets leader ip."""
-        return self.peers.leader_ip
-
```
Comment (dragomirp) [inline src/charm.py:992]: Not used anywhere

Comment (dragomirp) [inline src/charm.py:992]: Proxy function

File: src/relations/peers.py [hunk 2]
```diff
@@ -150,5 +150,5 @@ def _on_departed(self, _):
     def update_leader(self):
         """Updates leader hostname in peer databag to match this unit if it's the leader."""
         if self.charm.unit.is_leader():
-            self.charm.update_client_connection_info()
             self.app_databag[LEADER_ADDRESS_KEY] = self.charm.unit_ip
+            self.charm.update_client_connection_info()
```
Comment (dragomirp) [inline src/relations/peers.py:154]: IP is used in the connection info

**Review by taurus-forever — APPROVED** [review]
LGTM here.
Q: why did we notice this issue in pgbouncer repo?
As we have the same test with data-integrator here. Isn't it?

**Comment by dragomirp** [comment]
> LGTM here. Q: why did we notice this issue in pgbouncer repo? As we have the same test with data-integrator here. Isn't it?

I would guess different order of events, since the data-integrator tests have an active PG when they run the PGB tests.

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Pull request merged**